spec/acceptance/suites/rpm_docker/00_rpm_build_spec.rb

```diff
@@ -0,0 +1,120 @@
+#
+# If you set the environment variable BEAKER_copyin=yes will copy in your
+# 'simp-core' repo instead of using the one on the filesystem. You probably
+# want to also set BEAKER_destroy=no if you do this so that you can retrieve
+# any relevant artifacts.
+#
+#   * This mode could be **MUCH** slower but will preserve the sanctity of your
+#     workspace
+#
+require 'spec_helper_rpm'
+
+test_name 'RPM build'
+
+# Set build directory inside the container
+#
+# The /simp-core mount will be mounted regardless of the test requirements so
+# we have to use a different location if we want to copy things into the system
+#
+build_dir = ( ENV['BEAKER_copyin'] == 'yes' ) ? '/simp-import': '/simp-core'
+
+# Custom Gemfile Support
+gemfile = nil
+gemfile_path = File.expand_path(File.join(fixtures_path,'Gemfile'))
+
+if File.file?(gemfile_path)
+  gemfile = gemfile_path
+end
+
+describe 'RPM build' do
+  let(:local_basedir) { File.absolute_path(Dir.pwd) }
+
+  # We need a normal user for building the RPMs
+  let(:build_user) { 'build_user' }
+  let(:run_cmd) { %(runuser #{build_user} -l -c ) }
+
+
+  hosts.each do |host|
+    next if host[:roles].include?('disabled')
+
+    # This needs to happen *prior* to the test selection since the test
+    # selection depends on this being present
+    if ENV['BEAKER_copyin'] == 'yes'
+      %x(docker cp #{Dir.pwd} #{host.name}:#{build_dir})
+    end
+
+    it 'should clone the repo if necessary' do
+      set_perms = false
+
+      if ENV['BEAKER_copyin'] == 'yes'
+        set_perms = true
+      elsif !host.file_exist?("#{build_dir}/metadata.json")
+        # Handle Travis CI first
+        if ENV['TRAVIS_BUILD_DIR']
+          base_dir = File.dirname(ENV['TRAVIS_BUILD_DIR'])
+
+          %x(docker cp #{ENV['TRAVIS_BUILD_DIR']} #{host.name}:#{build_dir})
+
+          host.mkdir_p(base_dir)
+          on(host, %(cd #{base_dir}; ln -s #{build_dir} .))
+        else
+          # Just clone the main simp repo
+          on(host, %(git clone https://github.com/simp/simp-core #{build_dir}))
+        end
+
+        set_perms = true
+      end
+
+      if set_perms
+        on(host, %(chown -R #{build_user}:#{build_user} #{build_dir}))
+      end
+    end
+
+    it 'should have access to the local simp-core' do
+
+      # This is to work around irritating artifacts left around by r10k
+      unless local_basedir == build_dir
+        host.mkdir_p(File.dirname(local_basedir))
+
+        on(host, %(cd #{File.dirname(local_basedir)}; ln -s #{build_dir} #{File.basename(local_basedir)}))
+      end
+
+      host.file_exist?("#{local_basedir}/metadata.json")
+    end
+
+    it 'should align the build user uid and gid with the mounted filesystem' do
+      #on(host, %(groupmod -g `stat --printf="%g" #{build_dir}` #{build_user}))
+      on(host, %(if ! id `stat --printf="%g" #{build_dir}` >&/dev/null; then groupadd -g `stat --printf="%g" #{build_dir}` #{build_user}_supplementary; fi))
```

Comment:
This does not seem to work.  Maybe use getent group `stat --printf="%g" #{build_dir}`?   It is looking up the group id and then checking if a user exists with that id and then trying to create a group?